pkg/table/tables/index.go

```diff
@@ -206,10 +206,6 @@ func (c *index) create(sctx table.MutateContext, txn kv.Transaction, indexedValu
 		}
 
 		if untouched {
-			txn, err1 := sctx.Txn(true)
```

Comment:
We should not use `sctx.Txn()` here because it can be different with the txn in argument.

Reply:
Is there a test covering this?

Reply:
There is not bug now because untouched is always false in DDL reorg and `admin recover index` scene. But it is a potential risk...